pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -97,29 +100,44 @@ func (s *SelectorSpread) CalculateSpreadPriorityMap(pod *v1.Pod, meta interface{
 // where zone information is included on the nodes, it favors nodes
 // in zones with fewer existing matching pods.
 func (s *SelectorSpread) CalculateSpreadPriorityReduce(pod *v1.Pod, meta interface{}, nodeNameToInfo map[string]*schedulernodeinfo.NodeInfo, result schedulerapi.HostPriorityList) error {
+	var selectors []labels.Selector
+	priorityMeta, ok := meta.(*priorityMetadata)
+	if ok {
+		selectors = priorityMeta.podSelectors
+	} else {
+		selectors = getSelectors(pod, s.serviceLister, s.controllerLister, s.replicaSetLister, s.statefulSetLister)
+	}
 	countsByZone := make(map[string]int, 10)
 	maxCountByZone := int(0)
 	maxCountByNodeName := int(0)
 
-	for i := range result {
-		if result[i].Score > maxCountByNodeName {
-			maxCountByNodeName = result[i].Score
+	if len(selectors) > 0 {
+		pods, _ := s.podLister.Search(selectors)
+		if pods != nil && len(pods) > 0 {
```

Comment:
You can drop the nil check, because Go internally handles it in len() and will return 0 length.

Reply:
Done. Thanks.